### PR TITLE
D8 1385

### PR DESCRIPTION
**Pull request by protitude**
## Describe context / purpose for this PR

Add projects to community-persona pages

## Issue link

https://cyberteamportal.atlassian.net/jira/software/c/projects/D8/issues/D8-1385

## Any other related PRs?

https://github.com/necyberteam/cyberteam_drupal/pull/613

## Link to MultiDev instance

http://md-1385-accessmatch.pantheonsite.io

## Checklist for PR author
- [x] I have checked that the PR is ready to be merged
- [x] I have reviewed the DIFF and checked that the changes are as expected
- [x] I have assigned myself or someone else to review the PR

### CODE REVIEW

File: modules/cssn/src/Plugin/Util/ProjectLookup.php [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+      return;
+    }
+    foreach ($this->projects_sorted as $project) {
+      $stripe_class = $n % 2 == 0 ? 'bg-light' : '';
+      $title = $project['title'];
+      $sid = $project['sid'];
+      $project_status = $project['status'];
+      $project_name = $project['name'];
+      if (($project_status == 'Recruiting' && $project_name == 'Interested') || $project_name != 'Interested') {
+        $lowercase = lcfirst($project_name);
```
Comment (a-pasquale) [inline modules/cssn/src/Plugin/Util/ProjectLookup.php:134]: Seeing warnings like this:
```Deprecated function: lcfirst(): Passing null to parameter #1 ($string) of type string is deprecated in Drupal\cssn\Plugin\Util\ProjectLookup->getProjectList() (line 125 of modules/custom/access/modules/cssn/src/Plugin/Util/ProjectLookup.php).```
The null $project_name is also breaking the Font Awesome icon class.
The conditions on line 124 seem strange also. Seems like interested should be a status?

File: modules/cssn/src/Plugin/Util/ProjectLookup.php [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+      ];
+    }, $result_flag);
+    $result = array_merge($result, $flagged_results);
+    if ($result != NULL) {
+      foreach ($result as $project_result) {
+        $wf = WebformSubmission::load($project_result->sid);
+        $wf_lookup = $wf->getData();
+        $this->projects[] = [
+          'title' => $wf_lookup['project_title'],
+          'name' => $project_fields[$project_result->name],
```
Comment (a-pasquale) [inline modules/cssn/src/Plugin/Util/ProjectLookup.php:76]: Perhaps the issue in line 125 could be related to this warning? Not sure why $project_result->name would be "mentors" or "tags"?

```Warning: Undefined array key "mentors" in Drupal\cssn\Plugin\Util\ProjectLookup->runQuery() (line 68 of modules/custom/access/modules/cssn/src/Plugin/Util/ProjectLookup.php).
Warning: Undefined array key "tags" in Drupal\cssn\Plugin\Util\ProjectLookup->runQuery() (line 68 of modules/custom/access/modules/cssn/src/Plugin/Util/ProjectLookup.php).```

File: modules/cssn/src/Plugin/Util/ProjectLookup.php (hunk 1 above)
Comment (protitude) [reply modules/cssn/src/Plugin/Util/ProjectLookup.php:125]: I can't see any of these issues you're listing here. Are you testing as a specific user?

**Comment by a-pasquale** [comment]
"tags" is still causing errors. Not sure what "tags" is though.
https://md-1385-accessmatch.pantheonsite.io/community-persona/81

**Pull request merged**